mypy/semanal.py

```diff
@@ -4310,3 +4334,21 @@ def find_fixed_callable_return(expr: Expression) -> Optional[CallableType]:
             if isinstance(t.ret_type, CallableType):
                 return t.ret_type
     return None
+
+
+def unmark_any_as_explicit(t: Type) -> None:
```

Comment:
I think this name (and that of the associated class) needs a bit more thought -- I don't think it's clear what it means outside of the context of this PR.